ksqldb-common/src/main/java/io/confluent/ksql/util/ConsistencyOffsetVector.java

```diff
@@ -51,141 +51,95 @@
   // This field should be the first
   private int version;
   // Topic -> Partition -> Offset
-  private Map<String, Map<Integer, Long>> offsetVector;
-  private ReadWriteLock rwLock;
-
-  public ConsistencyOffsetVector() {
-    this.version = 0;
-    this.offsetVector = new HashMap<>();
-    this.rwLock = new ReentrantReadWriteLock();
-  }
+  private final ConcurrentHashMap<String, ConcurrentHashMap<Integer, Long>> offsetVector;
```

Comment:
Instead of copying all that from Position, why don't we just keep an private member of type Position and map all the operations?

Reply:
Do we need to keep the OffsetVector interface?

Reply:
I think we discussed this offline, unless you mean something else. `Position` does not have a public constructor so I cannot use the serialize/deserialize we currently have.

Reply:
Thanks, all, I'm not sure whether we need the OffsetVector interface or not (personally, I think not), but it's not a public API, so it's not crucially important to hammer that out.

Whether we keep a hashmap internally or a Position seems to be also up in the air; there are pros and cons on both sides. FWIW, I'm happier when there's less coupling in my life, so I'm more inclined toward the approach in this PR.

Reply:
Thanks both, I will keep the map for now